### DISCUSSION TITLE
Z-Up vs Y-Up

### DISCUSSION THREAD

**Discussion by AlexHolden** [zalo/CascadeStudio#133] [Q&A]
Hi, I just discovered Cascade Studio and really like it. One small issue I ran into: is there a simple way to export to an STL file with Z-up orientation instead of Y-up? I'm finding that a model that looks correct in the Cascade Studio CAD View pane gets flipped on its side when I export it as STL and import it into my CAM program (CamBam).

**Comment by raydeleu** [comment]
I confirmed that the STL export is rotated. When you export a shape as STL and import it again in CascadeStudio the complete shape is rotated around the x axis. A simple way to fix this is to rotate the shape before exporting. 

``` javascript 
shape = Rotate([1,0,0],90,shape);
```
![flipped-stl-2](https://user-images.githubusercontent.com/38007983/164836208-b7b88592-e6cd-4d46-9d04-e0113d7186c6.png)

The STEP file representation is oriented correctly.

**Reply by AlexHolden** [reply]
Thanks. What I ended up doing was rotating it after I imported it into my CAM program.

**Comment by raydeleu** [comment]
Browsing through the code I noticed that CascadeStudio uses one of the imported modules, "Three.js", to perform the export to STEP, STL and OBJ format. The Three.js viewer normally uses Y-Up. I guess that for all other exports this is corrected for but for the STL export this fails. See https://discourse.threejs.org/t/how-to-export-the-correct-stl-that-the-whole-file-is-horizontal/16343/3 for an approach to correct this in the Three.js viewer. I guess it is not simple to do this from within CascadeStudio.
